src/main/java/org/cloudfoundry/promregator/cfaccessor/ReactiveCFAccessorImpl.java

```diff
@@ -377,6 +389,30 @@ private void setupPaginatedRequestFetcher() {
 		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.SPACE_SUMMARY, spaceId, 
 				request, r -> this.cloudFoundryClient.spaces().getSummary(r), this.requestTimeoutAppSummary);
 
+  }
+
+	@Override
+	public Mono<ListOrganizationDomainsResponse> retrieveAllDomains(String orgId) {
+
+		ListOrganizationDomainsRequest request = ListOrganizationDomainsRequest.builder().organizationId(orgId).build();
+
+		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.APP_ROUTES, orgId, 
+		request, r -> this.cloudFoundryClient.organizations().listDomains(request), this.requestTimeoutAppRoutes);
 	}
 
+	@Override
+	public Mono<ListSpaceRoutesResponse> retrieveSpaceRoutes(String spaceId) {
+		ListSpaceRoutesRequest request = ListSpaceRoutesRequest.builder().spaceId(spaceId).build();
+
+		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.APP_ROUTES, spaceId, 
+		request, r -> this.cloudFoundryClient.spaces().listRoutes(request), this.requestTimeoutAppRoutes);
+	}
+
+	@Override
+	public Mono<ListRouteMappingsResponse> retrieveRouteMappings() {
+		ListRouteMappingsRequest request = ListRouteMappingsRequest.builder().build();
+
+		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.APP_ROUTES, "(empty)", 
+		request, r -> this.cloudFoundryClient.routeMappings().list(request), this.requestTimeoutAppRoutes);
```

Comment:
Wow. The response here might be very big, as we don't have any filtering here.
Even worse, it means that we will read all route mappings of all application on the entire installation. So, if we want to scrape 2 apps out of 10000 apps, then we still will have to store all 10000*n route mappings here.

Did you consider/estimate the memory consumption this could mean for us?

Reply:
As far as I can see this is the only way to do this with v2 without querying each app individually. There is no space or org level API for route_mappings.

It will _only_ be the route mappings you have the permissions to see (if that helps).

I didnt do much work on checking how this scales at load yet. However the route_mappings are quite small per mapping so im not sure even 10,000 would be _too_ challenging

Each route mapping is something like this

```
{
 "metadata": {
    "guid": "589bce61-48d3-476a-b90a-xxxxxxxxxxx",
    "url": "/v2/route_mappings/589bce61-48d3-476a-b90a-xxxxxxxxxxx",
    "created_at": "2021-02-01T12:16:24Z",
    "updated_at": "2021-02-01T12:16:24Z"
 },
 "entity": {
    "app_port": 80,
    "app_guid": "a802b42a-adb9-47c8-914c-xxxxxxxxxxx",
    "route_guid": "6b8ebb34-8ae1-4d1f-92a1-xxxxxxxxxxx",
    "app_url": "/v2/apps/a802b42a-adb9-47c8-914c-xxxxxxxxxxx",
    "route_url": "/v2/routes/6b8ebb34-8ae1-4d1f-92a1-xxxxxxxxxxx"
 }
```